src/TextInput/TextInput.features.stories.tsx

```diff
@@ -130,16 +130,24 @@ export const WithTrailingAction = () => {
       <FormControl>
         <FormControl.Label>Default label</FormControl.Label>
         <TextInput
+       
           trailingAction={
+            value.length > 0 ? (
             <TextInput.Action
               onClick={() => {
                 setValue('')
               }}
               icon={XCircleFillIcon}
               aria-label="Clear input"
-              sx={{color: 'fg.subtle'}}
-            />
-          }
+              sx={{
+                color: 'fg.subtle',
+                backgroundColor:'transparent',
+                '&:hover': {color: 'rgba(192, 192, 192, 2)'}
```

Comment:
I think for the hover state we should use `fg.subtle`. It's very subtle the change but it would be consistent usage of our tokens. 

I did consider using opacity but this could create accessibility contrast issues down the line.

Reply:
@maximedegreve indeed! what's your suggestion on this?

Reply:
Added them to the PR!